bridges/snowbridge/primitives/core/src/lib.rs

```diff
@@ -159,6 +159,12 @@ pub struct AssetMetadata {
 	pub decimals: u8,
 }
 
+impl AssetMetadata {
```

Comment:
Nonblocking: I would prefer to leave `AssetMetadata` as just plain old data and move this to the call site in the extrinsic. We do not re-use this `validate` method so its not needed. And I think we should add a test for the extrinsic to make sure the Error is raised in this case.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/183/commits/0808537318417bc57c08a4265e9d1464839fd361